app/controllers/api/vcs.php

```diff
@@ -0,0 +1,200 @@
+<?php
+
+use Utopia\App;
+use Appwrite\Event\Build;
+use Utopia\Database\Database;
+use Utopia\Database\Document;
+use Utopia\Database\Helpers\Permission;
+use Utopia\Database\Helpers\Role;
+use Appwrite\Utopia\Request;
+use Appwrite\Utopia\Response;
+use Utopia\Validator\Text;
+use Utopia\VCS\Adapter\Git\GitHub;
+use Utopia\Database\Helpers\ID;
+use Appwrite\Extend\Exception;
+use Utopia\Cache\Adapter\Redis;
+use Utopia\Cache\Cache;
+use Utopia\Database\Query;
+use Utopia\Database\Adapter\MariaDB;
+use Utopia\Database\Validator\Authorization;
+
+App::get('/v1/vcs/github/install')
+    ->desc('Install GitHub App')
+    ->groups(['api', 'vcs'])
+    ->label('scope', 'public')
+    ->label('origin', '*')
+    ->label('sdk.auth', [])
+    ->label('sdk.namespace', 'vcs')
+    ->label('sdk.method', 'install')
+    ->label('sdk.description', '')
+    ->label('sdk.response.code', Response::STATUS_CODE_MOVED_PERMANENTLY)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_HTML)
+    ->label('sdk.methodType', 'webAuth')
+    ->inject('response')
+    ->inject('project')
+    ->action(function (Response $response, Document $project) {
+        $projectId = $project->getId();
+        //TODO: Update the name of the GitHub App
+        $response->redirect("https://github.com/apps/demoappkh/installations/new?state=$projectId");
+    });
+
+App::get('/v1/vcs/github/setup')
+    ->desc('Capture installation id and state after GitHub App Installation')
+    ->groups(['api', 'vcs'])
+    ->label('scope', 'public')
+    ->param('installation_id', '', new Text(256), 'installation_id')
+    ->param('setup_action', '', new Text(256), 'setup_action')
+    ->param('state', '', new Text(256), 'state')
+    ->inject('response')
+    ->inject('dbForConsole')
+    ->action(function (string $installationId, string $setup_action, string $state, Response $response, Database $dbForConsole) {
+        //TODO: Fix the flow for updating GitHub installation
+
+        var_dump("hello1");
+        $project = $dbForConsole->getDocument('projects', $state);
+
+        var_dump($project);
+
+        var_dump($state);
+
+        if ($project->isEmpty()) {
+            throw new Exception(Exception::PROJECT_NOT_FOUND);
+        }
+
+        $projectInternalId = $project->getInternalId();
+
+        $github = new Document([
+            '$id' => ID::unique(),
+            '$permissions' => [
+                Permission::read(Role::any()),
+                Permission::update(Role::any()),
+                Permission::delete(Role::any()),
+            ],
+            'installationId' => $installationId,
+            'projectId' => $state,
+            'projectInternalId' => $projectInternalId,
+            'provider' => "GitHub",
+            'accessToken' => null
+        ]);
+
+        $github = $dbForConsole->createDocument('vcs', $github);
+
+        var_dump("hello");
+
+        $response
+            ->redirect("http://localhost:3000/console/project-$state/settings");
```

Comment:
I guess not ready yet, but we need to have a relative redirect. A good example can be found in the account controller.